### PR TITLE
Switch find_by_* to lookup_by_*

### PR DESCRIPTION
**Pull request by astrozzc**
The find_by_* methods defined in manageiq repo has conflicts with
rails dynamic finding and now they have been changed to lookup_by_*.
Switching the callers in this repo to the lookup_by_* methods.

Depend on https://github.com/ManageIQ/manageiq/pull/19313

@miq-bot add_label ivanchuk/no, changelog/yes, refactoring

### CODE REVIEW

**Comment by astrozzc** [comment]
@miq-bot add_label ivanchuk/no, changelog/yes, refactoring

**Comment by astrozzc** [comment]
@miq-bot add_reviewer @gmcculloug 
@miq-bot assign @tinaafitz

**Comment by astrozzc** [comment]
The Hakiri failure is not related to the code change here. It is requiring upgrade of rubyzip: https://hakiri.io/github/ManageIQ/manageiq-automation_engine/master/af8b61c86d6f7b2e5d968de90e1ea466aa693cad/warnings/10870eddf08c8f

File: lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb [hunk 1]
```diff
@@ -26,7 +26,8 @@ def self.respond_to_missing?(method_name, include_private = false)
 
     def self.allowed_find_method?(m)
       return false if m.starts_with?('find_or_create') || m.starts_with?('find_or_initialize')
-      m.starts_with?('find')
+
+      m.starts_with?('find', 'lookup_by')
```
Comment (gmcculloug) [inline lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb]: 👍

**Pull request merged**